platforms/documentation/docs/src/docs/userguide/dep-man/advanced/artifact_resolution.adoc

```diff
@@ -0,0 +1,356 @@
+
+== Artifacts
+
+An *artifact* is a file that is produced or consumed during the build process.
+*Artifacts* typically include compiled libraries, JAR files, AAR files, DLLs, or ZIPs.
+
+Let's look at the *metadata* for `org.jetbrains.kotlin:kotlin-stdlib:1.8.10` which showcases several *variants* and *artifacts*:
+
+[source,json]
+.kotlin-stdlib-1.8.10.module
+----
+{
+  "variants": [
+    {
+      "name": "apiElements",
+      "attributes": {
+        "org.gradle.category": "library",
+        "org.gradle.dependency.bundling": "external",
+        "org.gradle.jvm.version": "8",
+        "org.gradle.libraryelements": "jar",
+        "org.gradle.usage": "java-api"
+      },
+      "files": [
+        {
+          "name": "kotlin-stdlib-1.8.10.jar"
+        }
+      ]
+    },
+    {
+      "name": "runtimeElements",
+      "attributes": {
+        "org.gradle.category": "library",
+        "org.gradle.dependency.bundling": "external",
+        "org.gradle.jvm.version": "8",
+        "org.gradle.libraryelements": "jar",
+        "org.gradle.usage": "java-runtime"
+      },
+      "files": [
+        {
+          "name": "kotlin-stdlib-1.8.10.jar"
+        }
+      ]
+    },
+    {
+      "name": "jdk7ApiElements",
+      "attributes": {
+        "org.gradle.usage": "java-api",
+        "org.gradle.jvm.version": "7"
+      },
+      "files": [
+        {
+          "name": "kotlin-stdlib-jdk7-1.8.10.jar"
+        }
+      ]
+    },
+    {
+      "name": "jdk8ApiElements",
+      "attributes": {
+        "org.gradle.usage": "java-api",
+        "org.gradle.jvm.version": "8"
+      },
+      "files": [
+        {
+          "name": "kotlin-stdlib-jdk8-1.8.10.jar"
+        }
+      ]
+    }
+  ]
+}
+----
+
+As we can see there are a number of *artifacts* available in the *metadata*:
+
+[cols="1,1"]
+|===
+| Variant | Artifact(s) | Purpose
+
+| `apiElements`
+| `kotlin-stdlib-1.8.10.jar`
+| Standard Kotlin runtime (default).
+
+| `runtimeElements`
+| `kotlin-stdlib-jdk7-1.8.10.jar`
+| Provides additional APIs for Java 7 compatibility.
+
+| `jdk7ApiElements`
+| `kotlin-stdlib-jdk8-1.8.10.jar`
+| Provides additional APIs for Java 8 compatibility.
+
+| `jdk8ApiElements`
+| `kotlin-stdlib-common-1.8.10.jar`
+| Shared standard library for Kotlin Multiplatform.
+|===
+
+In an ideal case, once a *variant* is selected, the *artifact* should be automatically resolved.
+However, there are specific reasons why *artifact* selection still happens _after_ *variant* selection.
+For example, what if the metadata for `org.jetbrains.kotlin:kotlin-stdlib:1.8.10` looked like this:
+
+[source,json]
+.kotlin-stdlib-1.8.10.module
+----
+{
+  "variants": [
+    {
+      "name": "apiElements",
+      "attributes": {
+        "org.gradle.category": "library",
+        "org.gradle.dependency.bundling": "external",
+        "org.gradle.jvm.version": "8",
+        "org.gradle.libraryelements": "jar",
+        "org.gradle.usage": "java-api"
+      },
+      "files": [
```

Comment:
❗ I don't think this properly represents multiple Artifact Sets on a variant.  This is just a single variant with multiple artifacts.

Reply:
Fixed.